flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -449,7 +449,14 @@ public void addGeneralOptions(Options baseOptions) {
 	@Override
 	public AbstractYarnClusterDescriptor createClusterDescriptor(CommandLine commandLine) throws FlinkException {
 		final Configuration effectiveConfiguration = applyCommandLineOptionsToConfiguration(commandLine);
-
+		// for unit tests only
+		if (System.getenv("IN_TESTS") != null) {
```

Comment:
Instead of leaking the `IN_TESTS` hack out of the `AbstractYarnClusterDescriptor`, it would be better if the production and test code used the same code path. So for example, the `YarnTestBase` should setup the environment in such a way that the normal `FlinkYarnSessionCli` code finds the testing `YarnConfiguration`. That way, we would not have to insert special environment variables which signal that we are in a test.

An idea could be to modify the HADOOP_HOME env variable which is used when loading the Hadoop configuration. Maybe there is some other ENV variable which is respected by the YarnConfiguration which one could configure.

I think this would be a better solution to the problem. WDYT?